04-onion-routing.md

```diff
@@ -873,8 +931,54 @@ An _intermediate hop_ MUST NOT, but the _final node_:
   the final node's HTLC:
     - MUST return `final_incorrect_cltv_expiry` error.
   - if the `amt_to_forward` is greater than the `incoming_htlc_amt` from the
-  final node's HTLC:
+  final node's HTLC, AND `incomplete_payment` is not set:
     - MUST return a `final_incorrect_htlc_amount` error.
+  - if `incomplete_payment` is set:
+    - MUST use `amt_to_forward` to evaluate for
+      `incorrect_payment_amount` errors.
+    - SHOULD suspend incoming HTLCs:
+      - if the incoming HTLC has an `amt_to_forward` that is
+        different from existing suspended payments of the same
+        `payment_hash`:
+        - MUST fail the HTLC.
+        - MUST return an `incorrect_payment_amount` error.
+      - if the total `incoming_htlc_amt` of all suspended HTLCs and
```

Comment:
It is confusing. We are at the part where the amt_to_forward is different. Now you give an rational when to succeed the htlcs.

I would separate this somehow

Reply:
They are on the same level, so my mental model is that these rules are evaluated in that order, and this rule is not gated by the `amt_to_forward` rule.